tests/Algorand/SignerTests.cpp

```diff
@@ -86,3 +86,86 @@ TEST(AlgorandSigner, Sign) {
     ASSERT_EQ(hex(signature), "de73363dbdeda0682adca06f6268a16a6ec47253c94d5692dc1c49a84a05847812cf66d7c4cf07c7e2f50f143ec365d405e30b35117b264a994626054d2af604");
     ASSERT_EQ(hex(result), "82a3736967c440de73363dbdeda0682adca06f6268a16a6ec47253c94d5692dc1c49a84a05847812cf66d7c4cf07c7e2f50f143ec365d405e30b35117b264a994626054d2af604a374786e89a3616d74cd034fa3666565ce000775e3a2667633a367656eac6d61696e6e65742d76312e30a26768c420c061c4d8fc1dbdded2d7604be4568e3f6d041987ac37bde4b620b5ab39248adfa26c763da3726376c420a089aa6922e3b998fadff6cd4808ddf9e021e4944e389ea3d5c638786689197ea3736e64c42074b000b6368551a6066d713e2866002e8dab34b69ede09a72e85a39bbb1f7928a474797065a3706179");
 }
+
+TEST(AlgorandSigner, SignAsset) {
+    auto key = PrivateKey(parse_hex("c9d3cc16fecabe2747eab86b81528c6ed8b65efc1d6906d86aabc27187a1fe7c"));
+    auto publicKey = key.getPublicKey(TWPublicKeyTypeED25519);
+    auto from = Address(publicKey);
+    auto to = Address("UCE2U2JC4O4ZR6W763GUQCG57HQCDZEUJY4J5I6VYY4HQZUJDF7AKZO5GM");
+    Data note;
+    std::string genesisId = "mainnet-v1.0";
+    auto genesisHash = Base64::decode("wGHE2Pwdvd7S12BL5FaOP20EGYesN73ktiC1qzkkit8=");
+    auto transaction = AssetTransaction(
+        /* from */ from,
+        /* to */ to,
+        /* fee */ 488931,
+        /* amount */ 847,
+        /* asset id */13379146,
+        /* first round */ 51,
+        /* last round */ 61,
+        /* note */ note,
+        /* type */ "axfer",
+        /* genesis id*/ genesisId,
+        /* genesis hash*/ genesisHash
+    );
+
+    auto serialized = transaction.serialize();
+    auto signature = Signer::sign(key, transaction);
+    auto result = transaction.serialize(signature);
+
+    ASSERT_EQ(hex(serialized), "8aa461616d74cd034fa461726376c420a089aa6922e3b998fadff6cd4808ddf9e021e4944e389ea3d5c638786689197ea3666565ce000775e3a2667633a367656eac6d61696e6e65742d76312e30a26768c420c061c4d8fc1dbdded2d7604be4568e3f6d041987ac37bde4b620b5ab39248adfa26c763da3736e64c42074b000b6368551a6066d713e2866002e8dab34b69ede09a72e85a39bbb1f7928a474797065a56178666572a478616964ce00cc264a");
+    ASSERT_EQ(hex(signature), "9f9ab3c2a67e3b98c9db54e85a610fb61d7d4718e680a071f6fb7e9b037841faea432665a5838987bbec88ff3bf970d4512b7701016f88f114575f9ccf357a07");
+    ASSERT_EQ(hex(result), "82a3736967c4409f9ab3c2a67e3b98c9db54e85a610fb61d7d4718e680a071f6fb7e9b037841faea432665a5838987bbec88ff3bf970d4512b7701016f88f114575f9ccf357a07a374786e8aa461616d74cd034fa461726376c420a089aa6922e3b998fadff6cd4808ddf9e021e4944e389ea3d5c638786689197ea3666565ce000775e3a2667633a367656eac6d61696e6e65742d76312e30a26768c420c061c4d8fc1dbdded2d7604be4568e3f6d041987ac37bde4b620b5ab39248adfa26c763da3736e64c42074b000b6368551a6066d713e2866002e8dab34b69ede09a72e85a39bbb1f7928a474797065a56178666572a478616964ce00cc264a");
```

Comment:
could you please also add explorer link to this test tx?

Reply:
Not for this of course, here are very small rounds) I have stolen this data from other tests. But of course i have tested it on testnet. There are no similar data in other tests and tests look very synthetic, not real(because rounds are very small again). 
I have other transactions that I sent from iOS using this library assembled by me. Here they are:
https://testnet.algoexplorer.io/tx/K6OCFZ66YTXMKJTCWY4HWGEE6P6MQCWX6OKEH3GC5ICIX27HCRBA - opt-in transaction for USDC
https://testnet.algoexplorer.io/tx/2MYWI5ACWYUSVMDAALTG7LJUO53VKP6EKY5ZGGOPUTOFPRQZ5EMQ - transfer transaction of EURSA

Reply:
great, if possible could you please replicate them as tests?